src/ledger-iframe-bridge.ts

```diff
@@ -0,0 +1,291 @@
+import {
+  GetPublicKeyParams,
+  GetPublicKeyResponse,
+  LedgerBridge,
+  LedgerSignMessageParams,
+  LedgerSignMessageResponse,
+  LedgerSignTransactionParams,
+  LedgerSignTransactionResponse,
+  LedgerSignTypedDataParams,
+  LedgerSignTypedDataResponse,
+} from './ledger-bridge';
+
+const LEDGER_IFRAME_ID = 'LEDGER-IFRAME';
+
+export enum IFrameMessageAction {
+  LedgerConnectionChange = 'ledger-connection-change',
+  LedgerUnlock = 'ledger-unlock',
+  LedgerMakeApp = 'ledger-make-app',
+  LedgerUpdateTransport = 'ledger-update-transport',
+  LedgerSignTransaction = 'ledger-sign-transaction',
+  LedgerSignPersonalMessage = 'ledger-sign-personal-message',
+  LedgerSignTypedData = 'ledger-sign-typed-data',
+}
+
+type IFrameMessageResponse<TAction extends IFrameMessageAction> = {
```

Comment:
By representing return values for this type more accurately with union types. It allows TypeScript to infer return types from the `action` field.